### ISSUE TITLE
Catch API timeout errors

### ISSUE THREAD

**Issue by fra967** [dadi/web#86]
If API times-out for various reasons, Web should catch the error and return a 503 code

At present, it throws an error and restarts the service

Example error message
```
 Error: Datasource "xyz" failed. GATEWAY_TIMEOUT (504): 
    at DataHelper.processOutput (/dadi/web/node_modules/@dadi/web/dadi/lib/help.js:507:15)
    at IncomingMessage.<anonymous> (/dadi/web/node_modules/@dadi/web/dadi/lib/help.js:488:12)
    at emitNone (events.js:72:20)
    at IncomingMessage.emit (events.js:166:7)
    at endReadableNT (_stream_readable.js:913:12)
    at nextTickCallbackWith2Args (node.js:442:9)
    at process._tickDomainCallback (node.js:397:17)
[2016-08-19 10:46:48.578] [LOG]
[2016-08-19 10:46:48.578] [LOG]   Server stopped, process exiting...
error: Forever detected script exited with code: 0
error: Script restart attempt #1
```

**Comment by jimlambie** [comment]
No problem, have added to the 1.7.0 milestone. Version due for publish in the next week.

**Comment by jimlambie** [comment]
Re-reading this, I don't think that's the right way to do it. If a datasource fails due to an API timeout, why would we send a 503 back from _Web_?

I think a better way would be to attach the error to the returned data, and continue as normal?

**Comment by fra967** [comment]
it's ok, we could have a error handler in the web layer that shows a friendly message, as rendering won't be possible with missing data. The main thing is not to stop & the service.

**Comment by jimlambie** [comment]
@fra967 this has been added to v1.7.0 and also patched in 1.6. New patch version available now via NPM (v1.6.11)

A 503 or 504 response from API will return this block to the frontend (imagine a datasource named "movies"):

```
movies: {
  results: [],
  errors: [{
    code: 'WEB-0005',
    title: 'Datasource Timeout',
    details: 'The datasource "movies" timed out: ' Gateway Timeout (504):  http://api.example.com/1.0/db/movies?filter={}
  }]
}
```

**Comment by jimlambie** [comment]
Closed by https://github.com/dadi/web/commit/2bedd5200d29a3eef10dd785c79b5b2cedbc52ce
